src/features/web-compat.js

```diff
@@ -12,96 +11,97 @@ function windowSizingFix () {
 }
 
-/**
- * Add missing navigator.credentials API
- */
-function navigatorCredentialsFix () {
-    try {
-        if ('credentials' in navigator && 'get' in navigator.credentials) {
-            return
-        }
-        const value = {
-            get () {
-                return Promise.reject(new Error())
-            }
-        }
-        defineProperty(Navigator.prototype, 'credentials', {
-            value,
-            configurable: true,
-            enumerable: true
-        })
-    } catch {
-        // Ignore exceptions that could be caused by conflicting with other extensions
-    }
-}
-
-function safariObjectFix () {
-    try {
-        // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
-        if (window.safari) {
-            return
-        }
-        defineProperty(window, 'safari', {
-            value: {
-            },
-            configurable: true,
-            enumerable: true
-        })
-        // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
-        defineProperty(window.safari, 'pushNotification', {
-            value: {
-            },
-            configurable: true,
-            enumerable: true
-        })
-        // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
-        defineProperty(window.safari.pushNotification, 'toString', {
-            value: () => { return '[object SafariRemoteNotification]' },
-            configurable: true,
-            enumerable: true
-        })
-        class SafariRemoteNotificationPermission {
-            constructor () {
-                this.deviceToken = null
-                this.permission = 'denied'
-            }
-        }
-        // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
-        defineProperty(window.safari.pushNotification, 'permission', {
-            value: () => {
-                return new SafariRemoteNotificationPermission()
-            },
-            configurable: true,
-            enumerable: true
-        })
-        // @ts-expect-error https://app.asana.com/0/1201614831475344/1203979574128023/f
-        defineProperty(window.safari.pushNotification, 'requestPermission', {
-            value: (name, domain, options, callback) => {
-                if (typeof callback === 'function') {
-                    callback(new SafariRemoteNotificationPermission())
-                    return
-                }
-                const reason = "Invalid 'callback' value passed to safari.pushNotification.requestPermission(). Expected a function."
-                throw new Error(reason)
-            },
-            configurable: true,
-            enumerable: true
-        })
-    } catch {
-        // Ignore exceptions that could be caused by conflicting with other extensions
-    }
-}
-
+// TODO: verify that all descriptor shapes match the original (i.e. we're not overriding value descriptors with get/set descriptors)
```

Comment:
window.safari stuff is correct, navigator.credentials is wrong.

Change this comment to a TODO more local to the navigator.credentials.